test/powerlevel9k.spec

```diff
@@ -20,7 +20,7 @@ function testJoinedSegments() {
   POWERLEVEL9K_LEFT_PROMPT_ELEMENTS=(dir dir_joined)
   cd /tmp
 
-  assertEquals "%K{blue} %F{black}/tmp %K{blue}%F{black}%F{black}/tmp %k%F{blue}%f " "$(build_left_prompt)"
+  assertEquals "%K{blue} %F{black}/tmp%b %K{blue}%F{black}%F{black}/tmp%b %k%F{blue}%f " "$(build_left_prompt)"
```

Comment:
It looks like you've added bolding to all of the tests, is that right? We need to be sure that we are testing both bold & non-bold options. Really, what we care about is that the core functionality of the segment works, not that it works **bolded**. We can then test bolding separately without affecting all of the segment tests.

Reply:
I've added tests at the end to test the bolding and coloring

Reply:
All strings have `%b` as that doesn't really affect anything except being in the prompt string. Can code a way to remove it, but didn't think it was worth the increased code complexity. What do you think?

Reply:
Er, I'm confused. Let me ask a different question: why were the previous tests modified to add `%b` to the assertion strings? I must just not be understanding some aspect of this.

Reply:
Bolding is switched on with `%B` and off with `%b`. Since the `%b` doesn't affect the string, I declared `[[ POWERLEVEL9K_segment_BOLD == true ]] && bld='%B' || bld=''` in the code. I didn't do the same for bold off. So, `%b` *always* shows, but `%B` only shows when the segment is bold.

Make sense now?